src/components/MonacoContext.tsx

```diff
@@ -0,0 +1,30 @@
+import React, { createContext, useContext, useEffect, useState } from 'react';
+import monaco from '@monaco-editor/react';
+
+interface MonacoContextType {
+  monaco: typeof monaco | null;
+}
+
+const MonacoContext = createContext<MonacoContextType | undefined>(undefined);
+
+export const MonacoProvider: React.FC<{ children: React.ReactNode }> = ({ children }) => {
+  const [monacoInstance, setMonacoInstance] = useState<typeof monaco | null>(null);
+
+  useEffect(() => {
+    if (!monacoInstance) {
+      setMonacoInstance(monaco);
+    }
+  }, [monacoInstance]);
+
+  return (
+    <MonacoContext.Provider value={{ monaco: monacoInstance }}>{children}</MonacoContext.Provider>
+  );
+};
+
+export const useMonacoContext = (): MonacoContextType => {
+  const context = useContext(MonacoContext);
+  if (!context) {
+    throw new Error('useMonacoContext must be used within a MonacoProvider');
+  }
+  return context;
+};
```

Comment:
This file defines a MonacoContext and a useMonacoContext, but it seems they are not really used anywhere (besides wrapping the App with MonacoProvider).
What is their purpose? Can we get rid of them?
From the documentation of the editor it should be enough to import its React component and render it where needed, as you already did.